goon/browserassets/js/browserOutput.js

```diff
@@ -166,30 +151,76 @@ function emojiparse(el) {
+		var newElements = [];
+		var lastStop = 0;  // The end of the last found match.
+		// The loop is ugly.
+		// There are reasons for it, namely: BYOND does not support matchAll regex method, and the `regexp.exec()` which is
+		// "standard" workaround for older browsers does not respect sticky/global flags when matching .*, which
+		// means this thing never terminates.
+		while ((match = element.textContent.substring(lastStop).match(regex)) !== null) {
+			if (match[0].length == 0) {
+				// Matching on .* or other non-consuming pattern would get us here
+				break;
+			}
+			var start = match.index;
+			var newStop = match.index + match[0].length;
+			// push text between the end of the last match and start of this one
+			newElements.push(document.createTextNode(element.textContent.substring(lastStop, start)));
+			// wrap the match in a span, and push the span
+			var span = document.createElement("span");
+			span.style.background = opts.highlightColor;
+			span.appendChild(document.createTextNode(element.textContent.substring(start, newStop)));
+			newElements.push(span);
+			lastStop = newStop;
+		}
+		// Push the rest of the string.
+		newElements.push(document.createTextNode(element.textContent.substring(lastStop)));
+
+		// If we found any matches - replace the current text element with the bunch of new nodes we've just created
+		var lastInserted = element;
+		if (newElements.length > 0) {
+			for (var i = 0; i < newElements.length; i++){
+				lastInserted = element.parentNode.insertBefore(newElements[i], element);
+			}
+			element.parentNode.removeChild(element);
+		}
+		return lastInserted;
+	} else {
+		// If it's not a plain text node - recurse further.
+		var currentChild = element.childNodes[0];
+		while (currentChild){
+			currentChild = highlightRecursively(currentChild, term);
+			currentChild = currentChild.nextSibling;
+		}
+		return element;
+	}
 }
 
-//Highlights words based on user settings
+// Highlights words based on user settings
 function highlightTerms(el) {
-	if(regexHasError) return; //just stop right there ig the regex is gonna except
 	for (var i = 0; i < opts.highlightTerms.length; i++) { //Each highlight term
-		if(opts.highlightTerms[i]) {
-			if(!opts.highlightRegexEnable){
-				if(el.innerText.toString().toLowerCase().includes(opts.highlightTerms[i].toLowerCase())) //match normally
-				el.innerHTML = '<span style="background-color:'+opts.highlightColor+'">'+el.innerHTML+'</span>' //encloseincludes
-				continue;
-			}
-			var rexp;
-			try{
-				rexp = new RegExp(opts.highlightTerms[i],"gmi")
-			} catch(e){
-				el.innerHTML+='<br/><span style="boldwarning"> Your highlight regex - '+opts.highlightTerms[i]+' - is malformed. Thrown exception: '+e+'</span>'
-				regexHasError = true;
-				return;
+		if (opts.highlightTerms[i]) {
+			if (!opts.highlightRegexEnable) {
+				var innerTerms = opts.highlightTerms[i].split(" ")
```

Comment:
stolen from my pr, again

Reply:
I was basing this on your PR, correct. I didn't need to, but this is how it happened.